mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -73,15 +78,27 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
   @Since("1.4.0")
   def setOutputCol(value: String): this.type = set(outputCol, value)
 
+  /** @group setParam */
+  @Since("2.1.0")
+  def setHandleInvalid(value: String): this.type = set(handleInvalid, value)
+  setDefault(handleInvalid, "error")
+
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     transformSchema(dataset.schema)
-    val bucketizer = udf { feature: Double =>
-      Bucketizer.binarySearchForBuckets($(splits), feature)
+    val bucketizer: UserDefinedFunction = udf { (feature: Double, flag: String) =>
```

Comment:
Also, at this point, you already know the value for flag, so you can just use it here, rather than making the UDF take an extra argument.